usr.sbin/bsdinstall/scripts/cheri-desktop

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh
+#
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Copyright (c) 2022 Jessica Clarke <jrtc27@FreeBSD.org>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+
+############################################################ INCLUDES
+
+BSDCFG_SHARE="/usr/share/bsdconfig"
+. $BSDCFG_SHARE/common.subr || exit 1
+f_dprintf "%s: loading_includes..." "$0"
+f_include $BSDCFG_SHARE/dialog.subr
+
+############################################################ MAIN
+
+: ${BSDDIALOG_OK=0}
+: ${BSDDIALOG_CANCEL=1}
+
+bsddialog --backtitle "$OSNAME Installer" --title "CHERI Desktop" --yesno \
+    "Would you like to install the CHERI desktop environment?" 0 0
+if [ $? -ne $BSDDIALOG_OK ]; then
+	exit 0
+fi
+
+# Running before config stage so make sure chroot has working DNS
+f_quietly cp -f $BSDINSTALL_TMPETC/resolv.conf $BSDINSTALL_CHROOT/etc/
+
+clear
+echo "$OSNAME Installer"
+echo "========================"
+echo "CHERI Desktop"
+echo
+ASSUME_ALWAYS_YES=YES chroot $BSDINSTALL_CHROOT pkg64c install cheri-desktop
+# TODO: pkg64 install firefox (or a metapackage, ideally)
+# TODO: any loader FDT configuration needed
+
```

Comment:
Should we also add here:
```
sysrc -f $BSDINSTALL_CHROOT/etc/rc.conf cheri_desktop_enable="YES"
```
?

Reply:
No, it defaults to YES to reduce the number of steps users need to take if they don't install it during the installer (i.e. it's just "`pkg64c install cheri-desktop` and go" (plus probably a `pw groupmod video -m username`)).

Reply:
(and if we were to want to do that it'd be `echo cheri_desktop_enable=\"YES\" > $BSDINSTALL_TMPETC/rc.conf.cheri-desktop`)

Reply:
Would it be a good idea to inform a user about `pw groupmod` in a post-install package message?

Reply:
Maybe. seatd does already mention needing to be in the video group but the significance is probably lost on our users and I don’t think it’s the last thing printed, whereas for cheri-desktop it presumably would be, and we could make the message much more understandable. Though I suspect most will ignore it anyway; the getting started guide will need to document it either way.

Reply:
In order to reduce the risk users would miss that information, cheri-desktop could automatically add all reasonable users (with home directories in /home?) to the video group, list them and inform a user how to remove a user from the group with pw. It wouldn't be acceptable in upstream but probably most of our users wouldn't mind that.

Reply:
> No, it defaults to YES

I needed to do `service onestart sddm` in my case but it could be I was missing some changes in world.

Reply:
With which package set?

Reply:
> With which package set?

`service onestart sddm`? That was with packages just before https://github.com/CTSRD-CHERI/cheribsd-ports/pull/68.

Reply:
And you’d rebooted after installing them?

Reply:
Yes.

`service start sddm` works fine with the current world for me now.